### PR TITLE
feat(@angular/cli): adding resolver blueprints

### PR DESCRIPTION
**Pull request by sumitarora**
Fixes: https://github.com/angular/angular-cli/issues/6011

### CODE REVIEW

File: packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+
+@Injectable()
+export class <%= classifiedModuleName %>Resolver implements Resolve<any> {
+
+  constructor() { }
+
+  resolve(route: ActivatedRouteSnapshot) { }
```
Comment (clydin) [inline packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts]: The full signature is:
```
resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<T>|Promise<T>|T
```

Comment (sumitarora) [inline packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts]: @clydin done the changes

File: packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Rx';
```
Comment (cexbrayat) [inline packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts]: should be 'rxjs/Observable'

File: packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+@Injectable()
+export class <%= classifiedModuleName %>Resolver implements Resolve<any> {
+
+  constructor() { }
+
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any> {
```
Comment (cexbrayat) [inline packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts]: Maybe this should be like we did for guards and have the complete signature, i.e. `Observable<any> | Promise<any> | any` as return type

File: packages/@angular/cli/blueprints/resolver/index.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import { resolveModulePath } from '../../utilities/resolve-module-file';
+
+const Blueprint = require('../../ember-cli/lib/models/blueprint');
+const stringUtils = require('ember-cli-string-utils');
+const astUtils = require('../../utilities/ast-utils');
+const getFiles = Blueprint.prototype.files;
+
+export default Blueprint.extend({
+  description: '',
+  aliases: ['s'],
```
Comment (cexbrayat) [inline packages/@angular/cli/blueprints/resolver/index.ts]: should be `rs`

File: packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    return Observable.of(true);
```
Comment (cexbrayat) [inline packages/@angular/cli/blueprints/resolver/files/__path__/__name__.resolver.ts]: could only return `true` instead of using the `of` operator.

**Comment by sumitarora** [comment]
@cexbrayat Done with changes.

**Comment by cexbrayat** [comment]
@Brocco This was approved but never merged. Should that be reopened on `@angular/schematics`?

**Comment by hansl** [comment]
Closing every Blueprints related issue and PR.

**Pull request closed without merging**